version: "3.9"

volumes:
  authservicevolume:
    name: authservice-volume
networks:
  authservicenetwork:
    name: authservice-network
  internetservice:
    name: internet-service

services:
  authservice-postgres:
    container_name: auth-pg
    image: postgres:13
    restart: on-failure
    env_file:
      - ./auth/.env
    ports:
      - "${AUTH_POSTGRES_PORT}:${AUTH_POSTGRES_PORT}"
    volumes:
      - authservicevolume:/var/lib/postgresql/data
      - ./auth/database/scripts:/docker-entrypoint-initdb.d/
    networks:
      - authservicenetwork

  authservice-api:
    container_name: auth-api
    build: ./auth
    env_file:
      - ./auth/.env
    depends_on:
      - authservice-postgres
    volumes:
      - ./auth:/app
    ports:
      - "${AUTH_API_PORT}:${AUTH_API_PORT}"
    networks:
      - authservicenetwork
      - internetservice
    entrypoint: ["/bin/bash", "./scripts/entrypoint.dev.sh"]

  client:
    container_name: client
    build: ./client
    depends_on:
      - authservice-postgres
      - authservice-api
    ports:
      - "3000:3000"
    volumes:
      - ./client:/app
    networks:
      - internetservice
      #- /app/client/node_modules
